v2/workloadapi/client.go

```diff
@@ -249,37 +241,51 @@ func (c *Client) watchX509Context(ctx context.Context, watcher X509ContextWatche
 	ctx, cancel := context.WithCancel(withHeader(ctx))
 	defer cancel()
 
+	var err error
 	stream, err := c.wlClient.FetchX509SVID(ctx, &workload.X509SVIDRequest{})
 	if err != nil {
 		return err
 	}
 
 	for {
+		c.opts.log.Debugf("Watching X.509 contexts")
 		resp, err := stream.Recv()
 		if err != nil {
 			return err
 		}
 
 		svids := []*x509svid.SVID{}
 		bundles := []*x509bundle.Bundle{}
+		finish := true
```

Comment:
nit: instead of using a separate variable to track whether the response is ok, you could create a `func parseX509SVIDResponse(*workload.X509SVIDResponse) (*X509Context, error)`

Then the for loop gets real simple:
```
for {
    resp := recv()
    x509Context, err := parseX509SVIDResponse(resp)
    if err != nil {
        watcher.OnX509ContextWatchError(err)
        continue
    }
    watcher.OnX509ContextUpdate(x509Context)
}
```

Reply:
Much cleaner! I will make the changes.

Reply:
Just in case, I'm resetting the backoff after a successful read from the stream, but before parsing. Let me know if you think doing it after parsing is better.

```
for {
	resp, err := stream.Recv()
	if err != nil {
		return err
	}

	c.backoff.Reset()
	x509Context, err := parseX509SVIDResponse(resp)
	if err != nil {
		watcher.OnX509ContextWatchError(err)
		continue
	}
	watcher.OnX509ContextUpdate(x509Context)
}
```

Reply:
I think that looks right!

Reply:
Failure to parse the response won't tear down the stream so it shouldn't impact the backoff timing.